aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/MutationProcessorTest.java

```diff
@@ -221,4 +223,46 @@ public void conflictHandlerInvokedForUnhandledConflictError() throws DataStoreEx
         // Wait for the conflict handler to be called.
         Latch.await(handlerInvocationsRemainingCount);
     }
+
+    /**
+     * If the AppSync response to the mutation contains not-empty GraphQLResponse error
+     * list without any ConflictUnhandled error, then the user-provided error handler
+     * should be invoked.
+     * @throws DataStoreException On failure to obtain configuration from the provider
+     */
+    @Test
+    public void errorHandlerInvokedForError() throws DataStoreException {
+        // Arrange a user-provided error handler.
+        CountDownLatch handlerInvocationsRemainingCount = new CountDownLatch(1);
+        when(configurationProvider.getConfiguration())
+            .thenReturn(DataStoreConfiguration.builder()
+                .errorHandler(error -> handlerInvocationsRemainingCount.countDown())
```

Comment:
Right now, you test that the config provided was asked to supply an error handler. That's related to what you want to test, but not all the way there. _Ideally_, you would latch on the interior of an error handler closure, itself.

```console
errorToBeHandled -> {
    // Aha! The handler was actually invoked
}
```

Otherwise, this test seems pretty good!

Reply:
Luckily, the error handler is obtained from the configuration _ONLY_ when the handler is being invoked during runtime, so this is ultimately testing both haha. I can make it more explicit as you say though.

Reply:
Yea! The "explicit" part, that's all. Thanks :-D

Reply:
On second thought, this test is actually doing exactly that! The given lambda _is_ the implementation of error handler, so the fact that latch was counted down means the handler itself was invoked.